### DISCUSSION TITLE
Simple vault for passing sensitive variable

### DISCUSSION THREAD

**Discussion by kovalroma** [semaphoreui/semaphore#1428] [Ideas]
Having a simple secret manager or vault inside Semaphore would be nice. 
In my case, I need to pass some sensitive variables, like username, password, etc, to my Playbook.

For now, I'm using **Extra variables** feature. But it's now secure because everyone can have access to it.

**Comment by craigarms** [comment]
Hi,

You should store the vault in your repo and pass the var_file of your vault in the playbook, you then give your task template the vault password.

I don't think storing this in Semaphore is go pratice as you wouldn't be able to run your plays without semaphore.

**Comment by fiftin** [comment]
@kovalroma I think it is can be implemented. But require security research.
